src/Media_Command.php

```diff
@@ -372,10 +372,6 @@ private function needs_regeneration( $att_id, $fullsizepath ) {
 			return true;
 		}
 
-		if ( array_diff( get_intermediate_image_sizes(), array_keys( $metadata['sizes'] ) ) ) {
```

Comment:
Also, you'll need #4 when it lands so the tests actually run against your changes.